source/cl/test/UnitCL/include/kts/execution.h

```diff
@@ -347,6 +347,9 @@ class HalfParamExecution : public ExecutionWithParam<unsigned> {
   /// @brief Returns true if parameter at index has a scalar type
   /// @param[in] Index of parameter to check
   bool IsArgScalar(unsigned index) const;
+
+  /// @brief Returns a vector of edge cases that need extra testing.
+  virtual const std::vector<cl_ushort> &GetEdgeCases() const;
```

Comment:
is it really worth doing as a reference? It could be prone to people introducing bugs. Would it not be simpler to just add the edge cases on request?

Reply:
I don't quite get what you mean here, could you elaborate?

Reply:
I mean it can be called as is, it was more the reference, I'm just not sure that creating the vector at the time would really be noticeable time-wise

Reply:
True, this isn't performance-critical. I did check though that the risk you mentioned, of accidentally returning a local variable, is detected by the compiler. It is, it results in "error: reference to local variable ‘EdgeCases’ returned". It would be a warning, but we build with `-Werror` anyway. Because of that, I think the risk of accidental wrong use is low enough that we don't need to worry about it.

Reply:
I agree it does, feel free to merge as is.